remark-lint-nodejs-yaml-comments.js

```diff
@@ -41,10 +53,15 @@ function containsInvalidVersionNumber(version) {
     return version.some(containsInvalidVersionNumber);
   }
 
+  if (version?.[1] === "0" && (version[3] === "0" || version[3] === "1")) {
```

Comment:
Looks like this is Node 14+ only https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility so probably not good to use for cross version dep yet

Reply:
You're right, no need for this, it can be expressed even more cleanly using good old syntax. I've refactored that, PTAL.